server/src/main/java/org/elasticsearch/common/util/concurrent/ThrottledTaskRunner.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.util.concurrent;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.core.Releasable;
+
+import java.util.concurrent.Executor;
+
+public class ThrottledTaskRunner extends AbstractThrottledTaskRunner<ActionListener<Releasable>> {
```

Comment:
Is this used anywhere? If not, could we introduce it, when it is needed?

Reply:
I will be using it in https://github.com/elastic/elasticsearch/pull/92373 which is already quite a big change so I'd like to have prerequisites like this reviewed separately. It also came up in a discussion about search execution so I wanted to have the code somewhere visible. I'm happy to delay merging it until #92373 is closer to ready tho.